src/BSeries.jl

```diff
@@ -1254,33 +1256,231 @@ end
 # should create a lazy version, optionally a memoized one
 
 """
-    substitute(b, a, t::AbstractRootedTree)
+    TwoDerivativeRungeKuttaMethod(A1, b1, A2, b2, c = vec(sum(A1, dims=2)))
 
-Compute the coefficient corresponding to the tree `t` of the B-series that is
-formed by substituting the B-series `b` into the B-series `a`. It is assumed
-that the B-series `b` has the coefficient zero of the empty tree.
+Represent a two-derivative Runge-Kutta method with Butcher coefficients
+`A1`, `b1`, and `c` for the first derivative and `A2`, `b2` for the second
+derivative.
+If `c` is not provided, the usual "row sum" requirement of consistency with
+autonomous problems is applied.
 
-# References
+Given an ODE ``u'(t) = f(t, u(t))`` with ``u''(t) = g(t, u(t))``,
+one step from ``u^{n}`` to ``u^{n+1}`` is given by
+
+'''math
+\\begin{aligned}
+  y^i &= u^n + \\Delta t \\sum_j a^{1}_{i,j} f(t^n + c_i \\Delta t, y^i) + \\Delta t^2 \\sum_j a^{2}_{i,j} g(t^n + c_i \\Delta t, y^i), \\\
+  u^{n+1} &= u^n + \\Delta t \\sum_i b^1_{i} f(t^n + c_i \\Delta t, y^i) + \\Delta t^2 \\sum_i b^2_{i} g(t^n + c_i \\Delta t, y^i),
+\\end{aligned}
+'''
 
-Section 3.2 of
-- Philippe Chartier, Ernst Hairer, Gilles Vilmart (2010)
-  Algebraic Structures of B-series.
-  Foundations of Computational Mathematics
-  [DOI: 10.1007/s10208-010-9065-1](https://doi.org/10.1007/s10208-010-9065-1)
 """
-function substitute(b, a, t::AbstractRootedTree)
-    result = zero(first(values(a)) * first(values(b)))
+struct TwoDerivativeRungeKuttaMethod{T,
+                                     MatT <: AbstractMatrix{T},
+                                     VecT <: AbstractVector{T}} <:RootedTrees.AbstractTimeIntegrationMethod
+    A1::MatT
+    b1::VecT
+    c1::VecT
+    A2::MatT
+    b2::VecT
+    c2::VecT
```

Comment:
The documentation says we have only

```suggestion
    b1::VecT
    c::VecT
    A2::MatT
    b2::VecT
```

Reply:
In general c2 should always be zero, I'm not sure if its best to hard code that in, or still leave the option for it to be the row sum of A2